lib/beaker/hypervisor/openstack.rb

```diff
@@ -59,9 +59,23 @@ def initialize(openstack_hosts, options)
       @network_client ||= Fog::Network.new(networkoptionhash)
 
       if not @network_client
-
         raise "Unable to create OpenStack Network instance (api_key: #{@options[:openstack_api_key]}, username: #{@options[:openstack_username]}, auth_url: #{@options[:openstack_auth_url]}, tenant: #{@options[:openstack_tenant]})"
-                                                                                                                                      end
+      end
+
+      volumeoptionhash = {}
+      volumeoptionhash[:provider]           = :openstack
+      volumeoptionhash[:openstack_api_key]  = @options[:openstack_api_key]
+      volumeoptionhash[:openstack_username] = @options[:openstack_username]
+      volumeoptionhash[:openstack_auth_url] = @options[:openstack_auth_url]
+      volumeoptionhash[:openstack_tenant]   = @options[:openstack_tenant]
+      volumeoptionhash[:openstack_region]   = @options[:openstack_region] if @options[:openstack_region]
+
+      @volume_client ||= Fog::Volume.new(volumeoptionhash)
+
+      if not @volume_client
```

Comment:
You are forcing the creation of the volume_client, even though there may be no host['volumes'] to create.

Probably want to use the pattern:
```
def volume_client
  @volume_client ||= Fog::Volume.new(volumeoptionhash)
end
```